src/workers/parser/getOutOfScopeLocations.js

```diff
@@ -38,26 +42,72 @@ function getLocation(func) {
 }
 
 /**
- * Reduces an array of locations to remove items that are completely enclosed
- * by another location in the array.
+ * Find the nearest location containing the input bp position and
+ * return new locations without inner locations under that nearest location
+ *
+ * @param locations Notice! The locations MUST be sorted by `sortByStart`
+ *                  so that we can do linear time complexity operation.
  */
-function removeOverlaps(
-  locations: AstLocation | AstLocation[],
-  location: AstLocation
-) {
-  // support reducing without an initializing array
-  if (!Array.isArray(locations)) {
-    locations = [locations];
+function removeInnerLocations(locations: AstLocation[], position: AstPosition) {
+  // First, let's find the nearest bp-enclosing function location,
+  // which is to loop locations to find the last location enclosing the bp.
+  let nearestPos = -1;
```

Comment:
I think we can replace this logic w/ a lodash helper:

this attempt is likely flawed, but you should see what I have in mind :P 

```js
const firstInnerLocation = findIndex(locations,  loc => containsPosition(loc, position)
```

Reply:
@jasonLaster I was trying to import and utilize lodash. It works fine on the local page but while running the mochi test it fails because
```
JavaScript error: , line 0: uncaught exception: TypeError: _lodash is undefined
  JS Error: 
    "uncaught exception: TypeError: _lodash is undefined"]
```
Tried to search "lodash" under the mochi test folder bug found nothing to set. Did I miss some settings?

Reply:
oh yeah :) that's tricky. I forgot that this is running in a worker

so the typical way to use lodash is `import {x} from "lodash"` in the worker you need to do `import x from "lodash/x"` because we can't require lodash at runtime in the worker...